### ISSUE TITLE
Drop java 8 support (?)

### ISSUE THREAD

**Issue by Rollczi** [Rollczi/LiteCommands#375]

**Comment by huanmeng-qwq** [comment]
This is fatal for lower versions of Minecraft

**Comment by CarmJos** [comment]
Higher versions of Java have many pleasant features, but this is devastating to lower version games (minigames and PVP servers that still exist in large numbers).

**Comment by KermanIsPretty** [comment]
I think Lite Commands should update to atleast Java 17 following ViaVersion. Most Minigame/PvP servers already run a fork of the notchian server that can be updated or already support J17. Also if it is required lite commands can include a version that uses https://github.com/unimined/JvmDowngrader. I would love to see J21 support to utilize virtual threads, but its not a requirement.

**Comment by CarmJos** [comment]
> I think Lite Commands should update to atleast Java 17 following ViaVersion. Most Minigame/PvP servers already run a fork of the notchian server that can be updated or already support J17. Also if it is required lite commands can include a version that uses https://github.com/unimined/JvmDowngrader. I would love to see J21 support to utilize virtual threads, but its not a requirement.

Maybe you are right. But ProtocolLib says NO.

**Comment by Rollczi** [comment]
> > I think Lite Commands should update to atleast Java 17 following ViaVersion. Most Minigame/PvP servers already run a fork of the notchian server that can be updated or already support J17. Also if it is required lite commands can include a version that uses https://github.com/unimined/JvmDowngrader. I would love to see J21 support to utilize virtual threads, but its not a requirement.
> 
> Maybe you are right. But ProtocolLib says NO.

What do you mean?

**Comment by CarmJos** [comment]
> Owner Author

ProtocolLib for 1.8.8~1.12 alwasy drop errors if not using Java8 even using "Downgrader"

**Comment by CarmJos** [comment]
> > > I think Lite Commands should update to atleast Java 17 following ViaVersion. Most Minigame/PvP servers already run a fork of the notchian server that can be updated or already support J17. Also if it is required lite commands can include a version that uses https://github.com/unimined/JvmDowngrader. I would love to see J21 support to utilize virtual threads, but its not a requirement.
> > 
> > 
> > Maybe you are right. But ProtocolLib says NO.
> 
> What do you mean?

Of course, no matter what, I also strongly support LiteCommands upgrading to at least j17. 

The new version of Java brings many syntax and performance advantages.

**Comment by KermanIsPretty** [comment]
Any news on this? This would be an awesome change for the longevity of this project to update to at least Java 17. (21 preferred)

**Comment by Rollczi** [comment]
I'm in favor of this idea, but I'm currently involved in other functions

**Comment by Rollczi** [comment]
> > Owner Author
> 
> ProtocolLib for 1.8.8~1.12 alwasy drop errors if not using Java8 even using "Downgrader"

Do you know the cause or what the error looks like?

**Comment by Rollczi** [comment]
Okay, we probably don't want this solution. I see it could cause more problems than benefits. It's better if someone else uses JvmDowngrader in the final plugin.

**I'll be planning to move to Java 11.** 

It's recommended to use **modern Paper 1.8.8 forks**, which can even run on Java 21.
